### ISSUE TITLE
Unable to use custom certificate authority with Windows via WinRM

### ISSUE THREAD

**Issue by jd3nn1s** [hashicorp/terraform#16034]
### Terraform Version
Terraform v0.10.4-dev (f5a3dc09a7b74e3a0eb37774c46db2e4bcde17d2)

### Terraform Configuration Files
```hcl
resource "null_resource" "foo" {
  provisioner "file" {
    source = "test"
    destination = "c:/test"

    connection {
      type = "winrm"
      user = "Administrator"
      password = "xxxx"
      cacert = <<EOF
CA in PEM format goes here
EOF
      host = "34.209.11.11"
      https = true
      port = 5986
      timeout = "10s"
    }
  }
}
```

### Debug Output
https://gist.github.com/jd3nn1s/6b722a106e72e1382dd026e6c2ae5171

### Expected Behavior
Terraform uploads the test file via winrm

### Actual Behavior
Terraform fails to verify the certificate of the remote machine with the error "x509: certificate signed by unknown authority". 

This makes it impossible to establish a secure connection to a windows machine unless the certificate is trusted by a known certificate authority.

### Steps to Reproduce
1. Create a windows machine and configure with WinRM over TLS with a custom certificate
2. Paste the CA cert into test.tf
2. `terraform init`
3. `terraform apply`

### Important Factoids
Using a custom certificate authority

### References
#14783 resolves the issue. The provisioner is able to upload the file.

**Comment by jd3nn1s** [comment]
I'm unclear why this is labeled an enhancement rather than a bug. Terraform already claims to support specifying cacert for WinRM connections but the option does not work.

**Comment by jd3nn1s** [comment]
Fixed via merged #14783
